### ISSUE TITLE
[bug] NUnit2014 doesn't flag Enumerable.Any(predicate)

### ISSUE THREAD

**Issue by Bartleby2718** [nunit/nunit.analyzers#749]
Very similar to #748, but for [`.Any()`](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.any). NUnit.Analyzers should be able to convert
```cs
Assert.That(numbers.Any<int>(x => x % 2 == 0));
```
to
```cs
Assert.That(numbers, Has.Some.Matches<int>(x => x % 2 == 0));
```
but it doesn't.

**Comment by manfred-brands** [comment]
To call never asked for functionality a bug is a bit steep isn't it.

**Comment by Bartleby2718** [comment]
Well, [the description for NUnit2014 is "Use _SomeItemsConstraint_ for better assertion messages in case of failure"](https://docs.nunit.org/articles/nunit-analyzers/NUnit2014.html) as opposed to "Use _ContainConstraint_ for better assertion messages in case of failure." 🤷 

 I can change the wording if you believe this should not be part of NUnit2014 .

**Comment by manfred-brands** [comment]
If you are going to quote documentation, then be complete:
```markdown
Using `Does.Contain` (or `Does.Not.Contain`) constraint will lead to better assertion messages in case of failure,
so this analyzer marks all usages of string `Contains` method where it is possible to replace
with `Does.Contain` constraint.
```

Nothing in here about `Any` or `Matches`
